src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java

```diff
@@ -14,9 +15,10 @@
 
     @Override
     public User login(String login, String password) throws AuthenticationException {
-        Optional<User> optionalUser = userService.getByLogin(login);
-        if (optionalUser.isPresent() && optionalUser.get().getPassword().equals(password)) {
-            return optionalUser.get();
+        Optional<User> user = userService.getByLogin(login);
+        if (user.isPresent() && HashUtil.hashPassword(password, user.get().getSalt())
+                .equals(user.get().getPassword())) {
```

Comment:
let's create the private method `isPasswordValid(User user, String password)` to make this more readable

Reply:
done